### PR TITLE
OOP school library: set up associations 

### PR DESCRIPTION
**Pull request by anshuman7negi**
**In this branch, I:**

-  create a class Classroom.
-  Create the has-many/belongs-to relationship between Classroom and Student.
-  Create a class Book.
-  Create a class Rental.
-  Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,6 +1,8 @@
 require './person'
 
 class Student < Person
+  attr_accessor :classroom
+
```
Comment (TedLivist) [inline student.rb:5]: - [x] Kindly implement the method for the relationship in this student class.

  ```
  def classroom=(classroom)
    @classroom = classroom
    classroom.students.push(self) unless classroom.students.include?(self)
  end
  ```

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (TedLivist) [inline book.rb]: - [x] The `:rentals` should have a getter and setter (`attr_accessor`)

File: rental.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class Rental
+  attr_accessor :date
+  attr_reader :book, :person
```
Comment (TedLivist) [inline rental.rb]: - [x] The `:book` and `:person` attributes should have getter and setter too.

File: person.rb [hunk 4]
```diff
@@ -2,14 +2,20 @@
...
     @name = name
     @age = age
     @id = Random.rand(1..100)
+    @rentals = []
+  end
+
+  def rent_book(book, date)
+    rental = Rental.new(self, book, date)
+    rentals << rental
   end
```
Comment (TedLivist) [inline person.rb]: - [ ] Kindly note that creating a rental object on line 17 already adds the rental to the respective `rentals` arrays of book and person, so we don't need line 18.

- [x] Kindly also maintain the order of the arguments as defined in the Rental `initialize` method.

  ```
  def rent_book(book, date)
    Rental.new(date, book, self)
  end
  ```

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @anshuman7negi,

Good job so far!
✔️ Good implementation Rental class
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by afanimartin — APPROVED** [review]
Hi @anshuman7negi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Setter and Getter successfully added to the `Book` and `Rental` classes and `classroom` method added to the `Stident` class as suggested. Well done!

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

## Find me on slack
### Martin Afani

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**